src/transformers/generation_logits_process.py

```diff
@@ -614,3 +614,31 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         scores[scores == float("inf")] = torch.finfo(scores.dtype).max
 
         return scores
+
+
+class SoftLengthLogitsProcessor(LogitsProcessor):
```

Comment:
Not a huge fan of the name - it is in the end some kind of length penalty to me so could we maybe change it to something like:

```
ExponentialDecayLengthPenalty
```
?

Reply:
You won't believe how i struggled finding a proper name for the logitsprocessor. I like ExponentialDecayLengthPenalty, will change it accordingly.